symforce/slam/imu_preintegration/imu_factor.cc

```diff
@@ -0,0 +1,34 @@
+/* ----------------------------------------------------------------------------
+ * SymForce - Copyright 2022, Skydio, Inc.
+ * This source code is under the Apache 2.0 license found in the LICENSE file.
+ * ---------------------------------------------------------------------------- */
+
+#include "./imu_factor.h"
+
+#include <sym/factors/internal_imu_factor.h>
+
+namespace sym {
+
+template <typename Scalar>
+ImuFactor<Scalar>::ImuFactor(const PreintegratedImuMeasurements<Scalar>& pim)
+    : pim_{pim.pim}, sqrt_info_{pim.covariance.inverse().llt().matrixU()} {}
```

Comment:
Can we do something like `llt()` first and then triangular solve?
This is roughly 2 times faster if I recall correctly.

Reply:
Yeah it might be a bit of a premature optimization, but it does read as very strange to see `inverse().llt()`

Reply:
Well, inverting a triangular matrix makes a lot more sense than inverting a full square matrix, so I've done it.

Another note: It's actually fewer ops (goes from 6724 ops to 5964) to assume that the sqrt information matrix is lower triangular rather than upper triangular. This is because the residual is the error in `R`, then the error in `v`, then the error in `p`. However, `R` roughly only depends on rotational inputs, `v` on rotational + velocity & acceleration inputs, and `p` on everything, so, as a result, the `jacobian` of the residual is sparsest in the top rows and densest in the bottom rows.

Since an upper triangular matrix squashes the bottom rows into the top rows, we get a less sparse matrix than we would have had we used a lower triangular matrix (of course, if we wanted to use an upper trinagular matrix, we could simply reverse the order of `R`, `v`, and `p` and be just as efficient).